libtransmission/torrent-magnet.h

```diff
@@ -26,6 +27,22 @@ inline constexpr int METADATA_PIECE_SIZE = 1024 * 16;
 
 using tr_metadata_piece = small::max_size_vector<std::byte, 1024U * 16U>;
 
+struct metadata_node
+{
+    time_t requested_at = 0U;
+    int piece = 0;
+};
```

Comment:
If this is going to be in a public header, let's move its declaration inside of `tr_incomplete_metadata` so that we can make it private later & so we'll have one less name in the global namespace

Reply:
Good thinking. Ahh I was reluctant to move them into a header in the first place, but it looks unavoidable.

Reply:
I noticed this pattern happening when migrating other code from opaque C structures to C++ classes.

IMO this PR is an incremental step towards making tr_incomplete_metadata a proper class. Once all those methods in `torrent-magnet.cc` are member functions, then `metadata` and `pieces_needed` and `piece_count` and `struct metadata_node` can become private, just as they were opaque (public, but only visible to torrent-magnet.cc) before.